### PR TITLE
testing/xmrig: upgrade to 2.14.1, added aarch64, armv7 and armhf

### PR DESCRIPTION
**Pull request by OTLabs**
https://github.com/xmrig/xmrig/releases 2.14.0

### CODE REVIEW

**Comment by andypost** [comment]
btw it makes sense to try enable on arm arches https://github.com/xmrig/xmrig/pull/958

**Comment by OTLabs** [comment]
I already tried for aarch64 and got compilation error as some "asm/something.h" file was missing.

**Comment by OTLabs** [comment]
https://github.com/xmrig/xmrig/releases 2.14.1

**Comment by OTLabs** [comment]
aarch64 build fails due to lack of `<asm/hwcap.h>` header file. I will try to add linux-headers package.

armhf and armv7 builds fail due to: `gcc: error: unrecognized command line option '-maes'; did you mean '-mapcs'`.

**Comment by andypost** [comment]
Looks good, wondering the same header in musl-dev https://pkgs.alpinelinux.org/contents?file=hwcap.h&path=&name=&branch=edge&arch=aarch64

About other arm* I see https://github.com/xmrig/xmrig/issues/166

**Comment by OTLabs** [comment]
The header in musl-dev do not provide following definitions (compared to Linux v5.0):
```
#define HWCAP_SSBS		(1 << 28)
#define HWCAP_SB		(1 << 29)
#define HWCAP_PACA		(1 << 30)
#define HWCAP_PACG		(1UL << 31)
```
Ha, so `-maes` is an option for x86* arch only. I will take a look how to disable it. Maybe the architecture for armv7 and armhf is not correctly defined in drone, will take a look on this also.

**Comment by OTLabs** [comment]
Got armv7 compiling.

Expected values for arch were "armv7|armv7f|armv7s|armv7k|armv7-a|armv7l", reported value by drone is "armv8l" (it is an aarch64 machine in 32-bit mode).

**Comment by OTLabs** [comment]
Also got armhf.

File: testing/xmrig/APKBUILD [hunk 1]
```diff
@@ -2,22 +2,28 @@
...
+	armv7*)   ARM_TARGET=7; makedepends="$makedepends linux-headers" ;;
+	armhf*)   ARM_TARGET=7; makedepends="$makedepends linux-headers" ;;
+esac
+
 build() {
 	cd "$builddir"
 	mkdir build
 	cd build
-	cmake ..
+	cmake .. -DARM_TARGET=${ARM_TARGET}
```
Comment (andypost) [inline testing/xmrig/APKBUILD]: I'd better made it optional argument like `CMAKE_CROSSOPTS="-DARM_TARGET=8"` and used as `cmake .. ${CMAKE_CROSSOPTS}`

Comment (OTLabs) [reply testing/xmrig/APKBUILD]: Thank you for the tip! I will implement this tomorrow.

Comment (OTLabs) [reply testing/xmrig/APKBUILD]: Done!

**Comment by andypost** [comment]
Fixed on commit
```diff
-arch="x86_64 x86 aarch64 armv7 armhf"
+arch="all !s390x !ppc64le"
```

**Pull request closed without merging**